core/src/main/java/brooklyn/location/access/PortForwardManager.java

```diff
@@ -92,6 +144,16 @@
      * subsequently be looked up using {@link #lookup(String, int)}.
      */
     public void associate(String publicIpId, HostAndPort publicEndpoint, int privatePort);
+
+    /**
+     * Registers a listener, which will be notified each time a new port mapping is associated. See {@link #associate(String, HostAndPort, int)}
+     * and {@link #associate(String, HostAndPort, Location, int)}.
+     */
+    @Beta
+    public void addAssociationListener(AssociationListener listener, Predicate<? super AssociationMetadata> filter);
```

Comment:
Could do with a unit test in PortForwardManagerTest, that the `onAssociationCreated` and `onAssociationDeleted` methods get called.